.ci/publish.jenkins

```diff
@@ -0,0 +1,149 @@
+        String folderName = value
+        parallelJobs[folderName] = {
+            echo "Run parallel job for ${branchName} inside ${folderName}"
+            image.inside {
+                stage('Prepare sources') {
+                    writeJSON json: versions, file: "${folderName}/versions.json"
+                    if (folderName != 'latest') { 
+                        sh "cp -R latest/_themes ${folderName}/_themes"
+                    }
+                }
+
+                stage('HTML') {
+                    sh "sphinx-build -W -b html -d ${folderName}/_build/.doctrees ${folderName}/ gh-pages/en/${folderName}"
+                }
+
+                stage('PDF') {
+                    sh "sphinx-build -W -b latex -d ${folderName}/_build/.doctrees ${folderName}/ ${folderName}/_build/latex"
+                    sh "make -C ${folderName}/_build/latex all-pdf"
+                    sh "cp ${folderName}/_build/latex/conan.pdf gh-pages/en/${folderName}/conan.pdf"
+                }
+            }
+        }
+    }
+
+    stage('Generate all releases') {
+        parallelJobs.failFast = true
+        parallel parallelJobs
+    }
+
+    stage('Prepare gh-branch') {
+        dir('gh-pages') {
+            // sh 'cp src/versions.json versions.json'  // TODO: File is not used, remove from 'gh-pages' branch
+            sh 'cp en/latest/404.html 404.html'
+
+            String content = readFile('404.html')
+            String prefixLatest = "${prefix}en/latest"
+            content = content.replaceAll('href="_', "href=\"${prefixLatest}/_")
+            content = content.replaceAll('src="_', "src=\"${prefixLatest}/_")
+            content = content.replaceAll('alt="_', "alt=\"${prefixLatest}/_")
+            content = content.replaceAll('internal" href="', "internal\" href=\"${prefixLatest}/")
+            content = content.replaceAll('"search.html"', "\"${prefixLatest}/search.html\"")
+            content = content.replaceAll('"genindex.html"', "\"${prefixLatest}/genindex.html\"")
+            writeFile(file: '404.html', text: content)
+        }
+    }
+
+    stage('Archive generated folder') {
+        archiveArtifacts artifacts: 'gh-pages/**/*.*'
+        echo "Inspect generated webpage at ${BUILD_URL}artifact/gh-pages/index.html"
+    }
+
+    if (params.publish) {
+        stage('Publish to gh-pages') {
+            dir('gh-pages') {
+                sh 'git init .'
+                sh 'git checkout -b gh-pages'
+                sh 'git add .'
+                sh 'git config user.email "conanci@jfrog.com"'
+                sh 'git config user.name "ConanCI bot"'
+                sh 'git commit -m "Automatic deploy"'
+                //withCredentials([string(credentialsId: 'GITHUB_API_KEY', variable: 'GITHUB_API_KEY')]) {
+                //    sh "git remote add origin-pages https://${GITHUB_API_KEY}@github.com/conan-io/docs.git"
+                //}
+                //sh 'git push --quiet --force --set-upstream origin-pages gh-pages'
```

Comment:
This is commented intentionally, for the next PR, it is something I want to monitor during the first run